packages/hash/backend-utils/src/search/opensearch.ts

```diff
@@ -17,13 +28,80 @@ export type OpenSearchConfig = {
   maxConnectionAttempts?: number;
 };
 
+/**
+ * The OpenSearch pagination endpoint does not provide good context about progress.
+ * OpenSearch's opaque cursor is wrapped with context in order to supply more information to users.
+ * This for example allows for us to check if there there is a next page for a given query or not.
+ *
+ * Also used as an opaque cursor from the user's perspective.
+ */
+type OpenSearchCursorExtension = {
+  seenCount: number;
+  pageSize: number;
+  openSearchCursor: string;
+};
+
+const opaqueCursorToString = (params: OpenSearchCursorExtension): string =>
+  Buffer.from(JSON.stringify(params)).toString("base64");
+
+const opaqueCursorParse = (cursor: string): OpenSearchCursorExtension =>
+  JSON.parse(Buffer.from(cursor, "base64").toString());
+
+const generateSearchBody = (params: SearchParameters) => {
+  const fields = Object.entries(params.fields);
+  if (
+    fields.some(
+      ([_, it]) => typeof it.fuzziness === "number" && it.fuzziness < 0,
```

Comment:
checking it's not `"AUTO"` is technically equivalent to checking if it's a number since we rely on the typing, but more quickly/easily conveys all permitted values.
```suggestion
      ([_, it]) =>
        it.fuzziness !== "AUTO" &&
        (!Number.isInteger(it.fuzziness) || it.fuzziness < 0),
```